core/sql/sqlcomp/CmpSeabaseDDLtable.cpp

```diff
@@ -378,6 +378,244 @@ void CmpSeabaseDDL::createSeabaseTableLike(
   return;
 }
 
+// ----------------------------------------------------------------------------
+// Method: createSeabaseTableExternalHive
+//
+// This method creates a Trafodion table that represents a Hive table 
+//
+// in:
+//   cliInterface - references to the cli execution structure
+//   createTableNode - representation of the CREATE TABLE statement
+//   currCatName - catalog name to use, if not specified
+//   currSchName - schema name to use, if not specified
+//
+// returns:  0 - successful, -1 error, -2 error requiring cleanup
+//
+// any error detected is added to the diags area
+// ---------------------------------------------------------------------------- 
+short CmpSeabaseDDL::createSeabaseTableExternalHive(
+  ExeCliInterface &cliInterface,
+  StmtDDLCreateTable * createTableNode,
+  NAString &currCatName, 
+  NAString &currSchName)
+{
+  Lng32 retcode = 0;
+
+  ComObjectName srcTableName(createTableNode->getLikeSourceTableName(), COM_TABLE_NAME);
+  srcTableName.applyDefaults(currCatName, currSchName);
+
+  // For the target object, use TRAFODION as the default catalog and the source
+  // table's schema name as the default schema
+  ComObjectName tgtTableName(createTableNode->getTableName(), COM_TABLE_NAME);
+  ComAnsiNamePart currCatAnsiName(getSystemCatalog());
+  ComAnsiNamePart currSchAnsiName(srcTableName.getSchemaNamePart().getExternalName());
+  tgtTableName.applyDefaults(currCatName, currSchAnsiName);
+
+  // For now the object and schema name for the target table must match the
+  // object and schema name for the source table
+  if (tgtTableName.getObjectNamePart().getExternalName() != 
+      srcTableName.getObjectNamePart().getExternalName()) 
```

Comment:
In the JIRA I have a list of tasks left to be done.  One of them was testing with the latest version of HIVE which supports special characters. HBase, as you know, has a much more limited set of special characters.  So this will require some work and we may need to support a trafodion name different than the hive name.  Supporting a different name is also on the list of things left to do.